packages/simcore-sdk/tests/integration/test_node_ports_v2_r_clone.py

```diff
@@ -0,0 +1,180 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import shutil
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, AsyncIterable, Iterator
+
+import aioboto3
+import pytest
+import sqlalchemy as sa
+from faker import Faker
+from pytest_mock.plugin import MockerFixture
+from settings_library.r_clone import RCloneSettings
+from simcore_postgres_database.models.file_meta_data import file_meta_data
+from simcore_sdk.node_ports_v2 import r_clone
+
+pytest_simcore_core_services_selection = [
+    "migration",
+    "postgres",
+    "storage",
+]
+
+pytest_simcore_ops_services_selection = [
+    "minio",
+    "adminer",
+]
+
+
+class _TestException(Exception):
+    pass
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def file_name(faker: Faker) -> str:
+    return f"file_{faker.uuid4()}.txt"
+
+
+@pytest.fixture
+def upload_file_dir(tmpdir: Path) -> Iterator[Path]:
+    temp_path = Path(tmpdir)
+    assert temp_path.is_dir()
+    yield temp_path
+    shutil.rmtree(temp_path)
+
+
+@pytest.fixture
+def file_to_upload(upload_file_dir: Path, file_name: str, faker: Faker) -> Path:
+    # generate file with data
+    file_path = upload_file_dir / file_name
+    file_path.write_text(faker.paragraph(nb_sentences=5))
+    return file_path
+
+
+@pytest.fixture
+def local_file_for_download(upload_file_dir: Path, file_name: str) -> Path:
+    local_file_path = upload_file_dir / f"__local__{file_name}"
+    return local_file_path
+
+
+@pytest.fixture
+def s3_object(project_id: str, node_uuid: str, file_name: str) -> str:
+    s3_path = Path(project_id) / node_uuid / file_name
+    return f"{s3_path}"
+
+
+@pytest.fixture
+async def cleanup_s3(
+    r_clone_settings: RCloneSettings, s3_object: str
+) -> AsyncIterable[None]:
+    yield
+    async with _get_s3_object(r_clone_settings, s3_object) as s3_object:
+        await s3_object.delete()
+
+
+@pytest.fixture
+def mock_update_file_meta_data(mocker: MockerFixture) -> None:
+    async def _raise_error(*args, **kwargs) -> None:
+        raise _TestException()
+
+    mocker.patch(
+        "simcore_sdk.node_ports_v2.r_clone._update_file_meta_data",
+        side_effect=_raise_error,
```

Comment:
This works directly with ``side_effect=_TestException``

Reply:
What is the purpose of raising and exception when ``_update_file_meta_data`` is called? Why don't you return fake data instead?